### ISSUE TITLE
Warning for non-debug configuration

### ISSUE THREAD

**Issue by ecristerna** [stripe/stripe-ios#321]
This warning presents when testing the app in the debug config, using test cards.

> `Warning! You're building your app in a non-debug configuration, but appear to be using your Stripe test key. Make sure not to submit to the App Store with your test keys!`

How can I get rid of it?

**Comment by davidrothera** [comment]
What does your build config look like? I'm guessing this is because its looking at how you are compiling the target and it looks like a production build?

**Comment by ecristerna** [comment]
This is part of it. I have three build configs, where `DEVELOPMENT` is the one used for testing, and is where the warning shows

![screen shot 2016-04-06 at 9 29 59 am](https://cloud.githubusercontent.com/assets/3680271/14320353/62b35f72-fbda-11e5-803d-ba49c85b5519.png)

![screen shot 2016-04-06 at 9 30 19 am](https://cloud.githubusercontent.com/assets/3680271/14320357/6596b5cc-fbda-11e5-960e-abd6d24eb0e0.png)

**Comment by davidrothera** [comment]
Ok, that is setting the DEBUG macro so you can test for it in the binary but I am thinking what about the Optimization levels?

<img width="499" alt="screen shot 2016-04-06 at 15 42 07" src="https://cloud.githubusercontent.com/assets/2766321/14320759/2c9220d4-fc0e-11e5-8f2a-f3e744564a1a.png">

**Comment by davidrothera** [comment]
This is the code that is causing that message:

```obj-c
#ifndef DEBUG
    if ([publishableKey.lowercaseString hasPrefix:@"pk_test"]) {
        FAUXPAS_IGNORED_IN_METHOD(NSLogUsed);
        NSLog(@"⚠️ Warning! You're building your app in a non-debug configuration, but appear to be using your Stripe test key. Make sure not to submit to "
              @"the App Store with your test keys!⚠️");
    }
#endif
```

**Comment by davidrothera** [comment]
Your flags need to be:

<img width="104" alt="screen shot 2016-04-06 at 15 44 22" src="https://cloud.githubusercontent.com/assets/2766321/14320831/7616bf08-fc0e-11e5-8834-3aeb0e9749dd.png">

These definitely work (from my own app)

**Comment by ecristerna** [comment]
ok, so my flags are now set as yours, but still getting the warning.

this is the optimization level config...

![screen shot 2016-04-06 at 9 47 15 am](https://cloud.githubusercontent.com/assets/3680271/14320978/eb536b54-fbdc-11e5-93d8-e6295bc901ca.png)

**Comment by davidrothera** [comment]
How are you setting the key?

I use something like:

```swift
        #if DEBUG
            Stripe.setDefaultPublishableKey("pk_test_sldkjfsldkfjlsdkfj")
            log.debug("Debug mode enabled")
        #else
            Stripe.setDefaultPublishableKey("pk_live_lsdlkkskdjdfkjghdfk")
        #endif
```

**Comment by ecristerna** [comment]
same here, just the `DEVELOPMENT` macro instead of  your `DEBUG` one. I was wondering if it had something to do with the app written in Swift and Stripe in Ob-C, but I see yours is Swift also.

**Comment by davidrothera** [comment]
You don't have a `DEVELOPMENT` macro (according to your earlier screenshot), you only have `DEBUG`

**Comment by ecristerna** [comment]
right, just added it and still showing; the thing is the `DEBUG` macro is defined for muy `Development` configuration, so it shouldn't pass through the warning

```swift
#ifndef DEBUG
    if ([publishableKey.lowercaseString hasPrefix:@"pk_test"]) {
        FAUXPAS_IGNORED_IN_METHOD(NSLogUsed);
        NSLog(@"⚠️ Warning! You're building your app in a non-debug configuration, but appear to be using your Stripe test key. Make sure not to submit to "
              @"the App Store with your test keys!⚠️");
    }
#endif
```

**Comment by davidrothera** [comment]
Can you paste the code where you are setting your key>

**Comment by ecristerna** [comment]
```swift
        #if DEVELOPMENT
            Stripe.setDefaultPublishableKey("pk_test_m3NY5MSNaISIM2YPsgJgVw9P")
        #endif
        #if BETA
            Stripe.setDefaultPublishableKey("pk_test_m3NY5MSNaISIM2YPsgJgVw9P")
        #endif
        #if PRODUCTION
            Stripe.setDefaultPublishableKey("pk_live_pO6ElAmtNgNPdnsNpiXjZi5D")
        #endif
```

**Comment by davidrothera** [comment]
This isn't going to work, you aren't creating `DEVELOPMENT`, `BETA` or `PRODUCTION` macros, you are only defining the DEBUG macro which means you can only use DEBUG (like my example).

**Comment by ecristerna** [comment]
that code is working though, the keys are setting up correctly

**Comment by davidrothera** [comment]
After those lines can you do:

```swift
let key = Stripe.defaultPublishableKey()
print("Key: \(key)")
```

And see which it is hitting.

**Comment by ecristerna** [comment]
it's going in the first `if`

**Comment by jflinter** [comment]
Hey @ecristerna, I'm sorry for the slow reply on Stripe's behalf.

I suspect the conversation above is a red herring. Would you let me know how you're integrating our SDK? If you're using cocoapods, will you post the contents of your `Podfile.lock` here? If you're using Carthage, will you post the contents of your `Cartfile.resolved`? Thanks!

**Comment by jflinter** [comment]
(I've also updated this warning somewhat on the latest master, as there are cases where it'll be a false positive.)

**Comment by jflinter** [comment]
Closing this for the sake of issue triage, feel free to reply here still @ecristerna.

**Comment by ecristerna** [comment]
Hey @jflinter thanks for the reply,

I'm using Cooapods, and attaching the `Podfile.lock` file. I think I have the most recent version of the SDK, so the updates you mentioned did not make any difference.

[Podfile.lock.zip](https://github.com/stripe/stripe-ios/files/238812/Podfile.lock.zip)

**Comment by jflinter** [comment]
The update was on master, but we hadn't cut a release until this morning. If you update to the newly-released 7.0.0 this'll be resolved.
